var/spack/repos/builtin/packages/bacio/package.py

```diff
@@ -39,3 +39,6 @@ def cmake_args(self):
         args = [self.define_from_variant("CMAKE_POSITION_INDEPENDENT_CODE", "pic")]
 
         return args
+
+    def patch(self):
+        filter_file('2\.4\.0',str(self.spec.version),"VERSION",when="bacio@2.4.1")
```

Comment:
Version 2.4.0 is hardcoded?

Reply:
Yes, in the "VERSION" file in bacio repo